### DISCUSSION TITLE
Status of the "Recursive proofs" feature

### DISCUSSION THREAD

**Discussion by akoptelov** [MinaProtocol/mina#14104] [Q&A]
Hi, I would like to ask about the current status of the recursive/inductive proofs support in Mina. I'm aware of the experimental status of the feature, so it would be good to know if this is worth creating a proper issue.

Currently, when I'm trying to send a zkApp transaction with a method verifying an off-chain proof (like described [here](https://docs.minaprotocol.com/zkapps/o1js/recursion#recursively-verifying-a-linear-recursive-program-in-a-zkapp). this is the [code](https://gist.github.com/akoptelov/8e94e51f7fa1932e33b3ff2c3c2d47b8#file-recursion-ts)), I get an error ("Invalid rich scalar: Proof ..." in daemon the log, 500 from graphql endpoint).

I'm using snarkyjs 0.12.1 and 2.0.0rampup3 Mina daemon.

Another issue that I have is that I cannot work out a proper type for proof argument of a zkApp method. E.g. though the file mentioned above is transpiled to js properly by the `snarkyjs/run` script, this [line](https://gist.github.com/akoptelov/8e94e51f7fa1932e33b3ff2c3c2d47b8#file-recursion-ts-L52) makes type checker complain that 

> 'SimpleProgramProof' refers to a value, but is being used as a type here.

Also I cannot make this line pass type checking with e.g. tsc when using .ts files explicitly, e.g. with `ts-node`.

**Comment by akoptelov** [comment]
It looks like I managed to get rid of the first issue. There is no `sign` call for the transaction, so it is signed only by the zkapp privkey only. When I add a fee payer privkey as well, it is included into the chain, at least with 2.0.0rampup4 Mina version.

**Comment by L-as** [comment]
Ask in the Mina Discord in the zkApp section
